### PR TITLE
[SPARK-19009][DOC] Add streaming rest api doc

### PR DESCRIPTION
**Pull request by uncleGen**
## What changes were proposed in this pull request?

add streaming rest api doc

related to pr #16253

cc @saturday-shi @srowen

### CODE REVIEW

**Comment by uncleGen** [comment]
cc @srowen

**Comment by srowen** [comment]
I don't know about this change, so can't really review it. It's always best to ping people who wrote the original PR you're following on. @vanzin or @saturday-shi ?

**Comment by uncleGen** [comment]
cc @vanzin @saturday-shi

File: docs/monitoring.md [hunk 1]
```diff
@@ -312,15 +312,39 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
     <td>Details for the storage status of a given RDD.</td>
   </tr>
   <tr>
-    <td><code>/applications/[base-app-id]/logs</code></td>
-    <td>Download the event logs for all attempts of the given application as files within
+    <td><code>/applications/[app-id]/logs</code></td>
+    <td>Download the event logs of the given application as files within
```
Comment (vanzin) [inline docs/monitoring.md]: The previous version is more correct. It's also unrelated to streaming, so you probably shouldn't change this.

File: docs/monitoring.md [hunk 2]
```diff
@@ -312,15 +312,39 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
...
     a zip file.
     </td>
   </tr>
   <tr>
-    <td><code>/applications/[base-app-id]/[attempt-id]/logs</code></td>
```
Comment (vanzin) [inline docs/monitoring.md]: Don't remove this.

File: docs/monitoring.md [hunk 3]
```diff
@@ -312,15 +312,39 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
     <td>Details for the storage status of a given RDD.</td>
   </tr>
   <tr>
-    <td><code>/applications/[base-app-id]/logs</code></td>
```
Comment (vanzin) [inline docs/monitoring.md]: Don't change this. This is correct and matches the definition of `base-app-id` in the documentation.

File: docs/monitoring.md [hunk 4]
```diff
@@ -312,15 +312,39 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
...
   </tr>
   <tr>
-    <td><code>/applications/[base-app-id]/[attempt-id]/logs</code></td>
-    <td>Download the event logs for a specific application attempt as a zip file.</td>
+    <td><code>/applications/[app-id]/streaming/statistics</code></td>
+    <td>Statistics information of stream.</td>
   </tr>
+  <tr>
+    <td><code>/applications/[app-id]/streaming/receivers</code></td>
+    <td>A list of all receiver streams.</td>
```
Comment (vanzin) [inline docs/monitoring.md]: "receiver streams" -> "streaming receivers".

File: docs/monitoring.md [hunk 5]
```diff
@@ -312,15 +312,39 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
...
+    <td><code>/applications/[app-id]/logs</code></td>
+    <td>Download the event logs of the given application as files within
     a zip file.
     </td>
   </tr>
   <tr>
-    <td><code>/applications/[base-app-id]/[attempt-id]/logs</code></td>
-    <td>Download the event logs for a specific application attempt as a zip file.</td>
+    <td><code>/applications/[app-id]/streaming/statistics</code></td>
+    <td>Statistics information of stream.</td>
```
Comment (vanzin) [inline docs/monitoring.md]: "Statistics for the streaming context."

File: docs/monitoring.md [hunk 6]
```diff
@@ -312,15 +312,39 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
...
   </tr>
+  <tr>
+    <td><code>/applications/[app-id]/streaming/receivers</code></td>
+    <td>A list of all receiver streams.</td>
+  </tr>
+  <tr>
+    <td><code>/applications/[app-id]/streaming/receivers/[stream-id]</code></td>
+    <td>Details of the given receiver stream.</td>
```
Comment (vanzin) [inline docs/monitoring.md]: remove "stream".

**Comment by uncleGen** [comment]
@vanzin Take a review again please!

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**